src/code.cloudfoundry.org/policy-server/store/security_groups_store.go

```diff
@@ -0,0 +1,90 @@
+package store
+
+import "fmt"
+
+//counterfeiter:generate -o fakes/security_groups_store.go --fake-name SecurityGroupsStore . SecurityGroupsStore
+type SecurityGroupsStore interface {
+	Replace([]SecurityGroup) error
+}
+
+type SGStore struct {
+	Conn Database
+}
+
+func (sgs *SGStore) Replace(newSecurityGroups []SecurityGroup) error {
+	tx, err := sgs.Conn.Beginx()
+	if err != nil {
+		return fmt.Errorf("create transaction: %s", err)
+	}
+	defer tx.Rollback()
+
+	// delete all existing SGs and space associations
+	_, err = tx.Exec(tx.Rebind("DELETE from running_security_groups_spaces"))
+	if err != nil {
+		return fmt.Errorf("deleting running security group associations: %s", err)
+	}
+	_, err = tx.Exec(tx.Rebind("DELETE from staging_security_groups_spaces"))
+	if err != nil {
+		return fmt.Errorf("deleting staging security group associations: %s", err)
+	}
+	_, err = tx.Exec(tx.Rebind("DELETE from security_groups"))
+	if err != nil {
+		return fmt.Errorf("deleting security groups: %s", err)
+	}
+
+	// loop groups
+	for _, group := range newSecurityGroups {
```

Comment:
Worried if we'll be able to get through doing 3 inserts for *all* the security groups for a large installation every 60 seconds 😟